src/app/api/getDelegationsV2.ts

```diff
@@ -51,19 +54,25 @@ export const getDelegationsV2 = async (
   const delegationsAPIResponse: DelegationsAPIResponse = response.data;
 
   const delegations: DelegationV2[] = delegationsAPIResponse.data.map(
-    (apiDelegation: DelegationV2API): DelegationV2 => ({
-      finalityProviderBtcPksHex: apiDelegation.finality_provider_btc_pks_hex,
-      paramsVersion: apiDelegation.params_version,
-      stakerBtcPkHex: apiDelegation.staker_btc_pk_hex,
-      stakingAmount: apiDelegation.staking_amount,
-      stakingTime: apiDelegation.staking_time,
-      stakingTxHashHex: apiDelegation.staking_tx_hash_hex,
-      startHeight: apiDelegation.start_height,
-      endHeight: apiDelegation.end_height,
-      state: apiDelegation.state,
-      unbondingTime: apiDelegation.unbonding_time,
-      unbondingTx: apiDelegation.unbonding_tx,
-    }),
+    (apiDelegation: DelegationV2API): DelegationV2 => {
+      const state = DelegationV2StakingStateMap[apiDelegation.state];
+      if (!state) {
+        throw new Error(`Unknown delegation state: ${apiDelegation.state}`);
+      }
+      return {
+        finalityProviderBtcPksHex: apiDelegation.finality_provider_btc_pks_hex,
+        paramsVersion: parseInt(apiDelegation.params_version),
```

Comment:
hmm should we return string directly from the api side so we don't need to parse on the FE?